tabular/src/autogluon/tabular/models/xgboost/xgboost_model.py

```diff
@@ -0,0 +1,122 @@
+        for invalid in invalid_params:
+            if invalid in self.params:
+                self.params.pop(invalid)
+        params = self.params.copy()
+        max_category_levels = params.pop('proc.max_category_levels', 100)
+
+        verbosity = kwargs.get('verbosity', 2)
+        if verbosity <= 2:
+            verbose = False
+        elif verbosity == 3:
+            verbose = True
+            verbose_eval = 50
+        else:
+            verbose = True
+            verbose_eval = 1
+        
+        X_train = self.preprocess(X_train, is_train=True, max_category_levels=max_category_levels)
+        num_rows_train = X_train.shape[0]
+
+        eval_set = []
+        eval_metric = self.get_eval_metric()
+
+        if X_val is None:
+            early_stopping_rounds = 150
+            eval_set.append((X_train, y_train))  # TODO: if the train dataset is large, use sample of train dataset for validation
+        else:
+            modifier = 1 if num_rows_train <= 10000 else 10000 / num_rows_train
+            early_stopping_rounds = max(round(modifier * 150), 10)
+            X_val = self.preprocess(X_val, is_train=False)
+            eval_set.append((X_val, y_val))
+
+        try_import_xgboost()
+        from .callbacks import print_evaluation, early_stop_custom
+        callbacks = []
+        if verbose:
+            callbacks.append(print_evaluation(verbose_eval))
+        callbacks.append(early_stop_custom(early_stopping_rounds, start_time=start_time, time_limit=time_limit, verbose=verbose))
+
+        from xgboost import XGBClassifier, XGBRegressor
+        model_type = XGBClassifier if self.problem_type in PROBLEM_TYPES_CLASSIFICATION else XGBRegressor
+        self.model = model_type(**params)
+        self.model.fit(
+            X=X_train,
+            y=y_train,
+            eval_set=eval_set,
+            eval_metric=eval_metric,
+            verbose=False,
+            callbacks=callbacks
+        )
+
+        bst = self.model.get_booster()
+        self.params_trained['n_estimators'] = bst.best_iteration + 1
+        self.params_trained['best_ntree_limit'] = bst.best_ntree_limit
+
+    def get_model_feature_importance(self):
+        original_feature_names: list = self._ohe_generator.get_original_feature_names()
+        feature_names = self._ohe_generator.get_feature_names()
+        importances = self.model.feature_importances_.tolist()
+
+        importance_dict = {}
+        for original_feature in original_feature_names:
+            importance_dict[original_feature] = 0
+            for feature, value in zip(feature_names, importances):
+                if feature in self._ohe_generator.othercols:
+                    importance_dict[feature] = value
+                else:
+                    feature = '_'.join(feature.split('_')[:-1])
+                    if feature == original_feature:
+                        importance_dict[feature] += value
+
+        return importance_dict
```

Comment:
Can you add (requires rebase first):

```
from ...features.feature_metadata import R_OBJECT

    def _get_default_auxiliary_params(self) -> dict:
        default_auxiliary_params = super()._get_default_auxiliary_params()
        extra_auxiliary_params = dict(
            ignored_type_group_raw=[R_OBJECT],
        )
        default_auxiliary_params.update(extra_auxiliary_params)
        return default_auxiliary_params
```

This way the model will no longer crash if given object dtype input, and will just drop it instead. This is in preparation for multi-modal tabular+text support (#756)

Reply:
I really wanted the functions.